sql/core/src/main/scala/org/apache/spark/sql/execution/Queryable.scala

```diff
@@ -42,4 +43,67 @@ private[sql] trait Queryable {
   def explain(extended: Boolean): Unit
 
   def explain(): Unit
+
+  private[sql] def showString(_numRows: Int, truncate: Boolean = true): String
+
+  /**
+    * Format the string representing rows for output
+    * @param rows The rows to show
+    * @param numRows Number of rows to show
+    * @param hasMoreData Whether some rows are not shown due to the limit
+    * @param truncate Whether truncate long strings and align cells right
+    *
+    */
+  private[sql] def formatString (rows: Seq[Seq[String]],
```

Comment:
This indenting is wrong. https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide#SparkCodeStyleGuide-Indentation

4 space for function args that wrap.